File: TaskBoard.BLL/Models/ViewModels/Details/CardVmDetails.cs
using AutoMapper;
using TaskBoard.BLL.Common.Mapping;
using TaskBoard.BLL.Models.ViewModels.List;
using TaskBoard.DAL.Data.Entities;

namespace TaskBoard.BLL.Models.ViewModels.Details;

public class CardVmDetails : IMapWith<Card>
{
    public Guid Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime CreatedDate { get; set; }
    public DateTime DueDate { get; set; }
    public Guid PriorityId { get; set; }
    public string PriorityName { get; set; }
    public Guid CardListId { get; set; }
    public string CardListName { get; set; }
    public IReadOnlyList<HistoryLogVm> HistoryLogs { get; set; }

    public void Mapping(Profile profile)
    {
        profile.CreateMap<Card, CardVmDetails>()
            .ForMember(dest => dest.PriorityName, opt => opt.MapFrom(src => src.Priority.Name))
            .ForMember(dest => dest.CardListName, opt => opt.MapFrom(src => src.CardList.Name));
    }
}

File: TaskBoard.UI/src/app/models/view-models/card-vm-details.ts
import {HistorylogVm} from "./historylog-vm";

export class CardVmDetails{
  id= '';
  name= '';
  description= '';
  createdDate: Date = new Date();
  dueDate: Date = new Date();
  priorityId= '';
  priorityName= '';
  cardListId= '';
  cardListName= '';
  historyLogs: HistorylogVm[] = [];
}
